### ISSUE TITLE
Problem with actionother in AutoRespond

### ISSUE THREAD

**Issue by ipadalert** [ORelio/Minecraft-Console-Client#712]
To just get to the point when using actionother in my matches I set it up so it can detect and respond to faction messages so if someone who wasn't me wanted it to accept a tp and make it sethome they can without myself having to do it. This works fine of course when my factions members make it for example accept a tp or something however, if the exact line were to be said in private messages it could say something or do something not by someone who I allowed. I just wanted to ask if there was a way to detect one line and make sure there is nothing before it to make sure it is coming from faction chat and not someone else. 
To clafiry things:
![Screen Shot 2019-04-27 at 12 12 47 PM](https://user-images.githubusercontent.com/45241936/56852192-fe043980-68e5-11e9-9fd5-75d7fdd0bcef.png)
![Screen Shot 2019-04-27 at 12 13 29 PM](https://user-images.githubusercontent.com/45241936/56852193-fe043980-68e5-11e9-9915-eb47970cc90a.png)
![Screen Shot 2019-04-27 at 12 13 57 PM](https://user-images.githubusercontent.com/45241936/56852194-fe043980-68e5-11e9-900e-71709d57fb42.png)
So basically people can just resay the match in pms or something and it will respond which is not good at all and I wanted to know if there is anyway to make matches detect and respond to only on the start of the line and not with everything before it.

**Comment by ORelio** [comment]
You need to improve your regex so that it does not match _anything_ with `(.+)`.
I'd suggest `^[^]]+ \*\*ipadalert: go f c f` for instance, to skip the header and resume at message start.

**Comment by ipadalert** [comment]
Even with a different regex it still does the same thing, on a different server I have it with a match instead of a regex and it still responds to it.
![Screen Shot 2019-04-27 at 3 55 11 PM](https://user-images.githubusercontent.com/45241936/56854464-dcff1100-6904-11e9-8d4f-a098cb5676c3.png)
![Screen Shot 2019-04-27 at 3 54 45 PM](https://user-images.githubusercontent.com/45241936/56854466-de303e00-6904-11e9-9723-ae0f641a6e20.png)
![Screen Shot 2019-04-27 at 3 52 53 PM](https://user-images.githubusercontent.com/45241936/56854467-dffa0180-6904-11e9-865d-1635fdc0c355.png)

**Comment by ORelio** [comment]
1. Go to https://regex101.com/
2. In "Test String", put example messages that you want the regex to match and some you _don't_
3. Make a regex that match the ones you want, and not the other ones
4. Or even better, create [ChatFormat] regexes in config that work for your server, and use `actionprivate` and `ownersonly=true`, then set your username as bot owner

**Comment by ipadalert** [comment]
Yeah I've been using https://regex101.com/ to test these but even then there could be a message the basically clones what the match is when messages to the account. I already have actionprivate and actionpublic used for my chat formats so there is no work around for faction chat. Along with both private and public having the same issues as action other as you can see here: 
![Screen Shot 2019-04-27 at 4 06 42 PM](https://user-images.githubusercontent.com/45241936/56854552-7975e300-6906-11e9-830d-e922062e1571.png)
![Screen Shot 2019-04-27 at 4 06 28 PM](https://user-images.githubusercontent.com/45241936/56854553-7975e300-6906-11e9-96fd-a66b49d42057.png)
![Screen Shot 2019-04-27 at 4 04 58 PM](https://user-images.githubusercontent.com/45241936/56854554-7975e300-6906-11e9-9a9b-76abbb73e87b.png)
I have used bot owners once but it didn't match anything from my matches.ini and so I just left it be for a while but I will test it again.

**Comment by ipadalert** [comment]
Owners doesn't work for actionother so it somewhat works but if someone were to message the account cloning what faction chat said they be able to do things to the account that I don't want.

**Comment by ORelio** [comment]
In regex, you need to put `^` at beginning to specify start of line, otherwise it will start matching anywhere inside the line. Same with `$` for end of line. So the regex should look like this: `^pattern$`

**Comment by ipadalert** [comment]
Yeah it worked now, thanks for helping out. Really sorry if I wasted any of your time, kind of a stupid mistake I made.

**Comment by ORelio** [comment]
Regex is not easy to learn, unfortunately. But glad you got it working.
